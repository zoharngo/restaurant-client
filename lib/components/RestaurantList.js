import React, { Component } from 'react';
import Restaurant from './Restaurant';
import RestaurantService from 'restaurant-service';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

const restaurantService = new RestaurantService();

export default class RestaurantList extends Component {

  constructor(props) {
    super(props);

    this.state = {
      ...props
    };
  }

  static contextTypes = {
    store: PropTypes.object,
  };

  removeRestaurant = (uuid) => {
    restaurantService.deleteRestaurant(uuid).then(() => {
      this.context.store.removeStateElement(uuid);
    });
  };

  filterRestaurants = (e) => {
    const restaurants = Object.assign({},  this.context.store.getState().restaurants);
    const searchTxt = e.currentTarget.value.toUpperCase();
    if (searchTxt) {
      Object.values(restaurants).filter((rest) => {
        if (rest['restaurant_name'].toUpperCase().indexOf(searchTxt) == -1)
          delete restaurants[rest['uuid']];
      });
    }
    this.setState(() => ({ restaurants: restaurants }));
  }

  render() {
    const { restaurants } = this.state;
    return (
      <div>
        <input style={styles.searchBar} className="form-control" type="text" placeholder="Search" onChange={(e) => this.filterRestaurants(e)} />
        <ul className="list-group" style={styles.list}>
          <li className="list-group-item">
            {Object.values(restaurants).map((rest) =>
              (<div style={styles.restaurant} key={rest.uuid} >
                <Restaurant style={styles.restaurant} restaurant={rest} />
                <div style={styles.buttons}>
                  <Button className="btn btn-warning" value={rest.uuid} style={styles.button} onClick={() => this.removeRestaurant(rest.uuid)}>Delete</Button>
                </div>
              </div>)
            )}
          </li>
        </ul>
      </div>

    );
  }
}


const styles = {
  searchBar: {
    marginBottom: 10,
  },
  restaurant: {
    borderBottomStyle: 'solid',
    borderBottomColor: '#aaa',
    borderBottomWidth: 1,
    paddingBottom: 10,
    paddingTop: 10,
  },
  list: {
    borderTopStyle: 'solid',
    borderTopColor: '#aaa',
    borderTopWidth: 1,
  },
  buttons: {
    marginTop: 10,
    paddingTop: 2,
    paddingBottom: 2,
  },
  button: {
    padding: 2,
    marginRight: 5,
    width: 50,
  },
};



