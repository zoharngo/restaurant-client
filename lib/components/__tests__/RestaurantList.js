import React from 'react';
import RestaurantList from '../RestaurantList';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import StoreApi from 'state-api';

describe('RestaurantList', () => {


  const store = new StoreApi([{ uuid: 'a' }, { uuid: 'b' }]);
  const testProps = { store: store };

  it('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <RestaurantList
          {...testProps}
        />
      </Router>

    ).toJSON();
    console.info(tree);
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });

});