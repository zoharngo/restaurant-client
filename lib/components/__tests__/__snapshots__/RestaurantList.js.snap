// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList renders correctly 1`] = `
<div>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search"
    style={
      Object {
        "marginBottom": 10,
      }
    }
    type="text"
  />
  <ul
    className="list-group"
    style={
      Object {
        "borderTopColor": "#aaa",
        "borderTopStyle": "solid",
        "borderTopWidth": 1,
      }
    }
  >
    <li
      className="list-group-item"
    >
      <div
        style={
          Object {
            "borderBottomColor": "#aaa",
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            <a
              className="btn btn-link"
              href="/RestaurantEdit/a"
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                  "fontSize": 18,
                }
              }
            >
              Edit
            </a>
          </div>
          <div
            style={
              Object {
                "color": "#888",
                "fontSize": "0.95em",
              }
            }
          />
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          />
          <div>
            <span />
          </div>
        </div>
        <div
          style={
            Object {
              "marginTop": 10,
              "paddingBottom": 2,
              "paddingTop": 2,
            }
          }
        >
          <button
            className="btn btn-warning btn btn-default"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "marginRight": 5,
                "padding": 2,
                "width": 50,
              }
            }
            type="button"
            value="a"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "borderBottomColor": "#aaa",
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            <a
              className="btn btn-link"
              href="/RestaurantEdit/b"
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                  "fontSize": 18,
                }
              }
            >
              Edit
            </a>
          </div>
          <div
            style={
              Object {
                "color": "#888",
                "fontSize": "0.95em",
              }
            }
          />
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          />
          <div>
            <span />
          </div>
        </div>
        <div
          style={
            Object {
              "marginTop": 10,
              "paddingBottom": 2,
              "paddingTop": 2,
            }
          }
        >
          <button
            className="btn btn-warning btn btn-default"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "marginRight": 5,
                "padding": 2,
                "width": 50,
              }
            }
            type="button"
            value="b"
          >
            Delete
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
